src/arrayMethodJoin.js

```diff
@@ -5,7 +5,42 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const arrCopy = [...this];
+    let string = '';
+
+    if (arrCopy.length === 0) {
+      return string;
+    }
+
+    for (let i = 0; i < arrCopy.length; i++) {
+      if (arrCopy[i] === null || arrCopy[i] === undefined) {
+        arrCopy[i] = '';
+      }
+    }
+
+    if (separator || separator === null) {
+      for (let i = 0; i < arrCopy.length - 1; i++) {
+        string = string + arrCopy[i] + separator;
+      }
+      string = string + arrCopy[arrCopy.length - 1];
+
+      return string;
+    }
+
+    if (separator === '' || separator === null) {
+      for (let i = 0; i < arrCopy.length; i++) {
+        string = string + arrCopy[i];
+      }
+
+      return string;
+    }
+
+    for (let i = 0; i < arrCopy.length - 1; i++) {
+      string = string + arrCopy[i] + ',';
+    }
+    string = string + arrCopy[arrCopy.length - 1];
```

Comment:
No need to do so many operations. Algoritm should be next
create for loop, inside there are 2 if statement
one `if` statement check that this[i] not equal null or undefined if so - add to result `string += this[i]`
second `if` check if i is not last and add separator